### ISSUE TITLE
can2mqtt stops after some readings

### ISSUE THREAD

**Issue by maramend** [Hunv/can2mqtt#1]
Hi,

I setup your `can2mqtt` on a Raspberry Pi 4B with a current Raspberry Pi OS Lite.

`Linux raspberrypi 5.10.17-v7l+ #1403 SMP Mon Feb 22 11:33:35 GMT 2021 armv7l GNU/Linux`

I followed your intructions to install the `can2mqtt ` and it starts to transmitt the received CAN frames to my local MQTT broker (127.0.0.1), where I can receive the MQTT messages.

```
/usr/local/bin/dotnet /home/pi/can2mqtt_core/can2mqtt_core.dll --Daemon:MqttServer="127.0.0.1" --Daemon:MqttTranslator="StiebelEltron"
info: can2mqtt_core.DaemonService[0]
      Starting: Can2Mqtt
Application started. Press Ctrl+C to shut down.
Hosting environment: Production
Content root path: /home/pi/can2mqtt_core/
CONNECTED TO MQTT BROKER 127.0.0.1 using ClientId Can2Mqtt
CONNECTED TO CANLOGSERVER 127.0.0.1 ON PORT 28700
Received CAN Frame: (1616494454.434776) can0 180#2200FA4F4600AE
Sending MQTT Message: 180#2200FA4F4600AE and Topic Can2Mqtt
Received CAN Frame: (1616494454.558405) can0 100#3100FA4F410000
Received CAN Frame: (1616494454.564273) can0 180#2200FA4F410154
Sending MQTT Message: 180#2200FA4F410154 and Topic Can2Mqtt
Received CAN Frame: (1616494454.687213) can0 100#3100FA4F430000
Received CAN Frame: (1616494454.693692) can0 180#2200FA4F43014C
Sending MQTT Message: 180#2200FA4F43014C and Topic Can2Mqtt
Received CAN Frame: (1616494455.799755) can0 500#E100FAFE070000
Received CAN Frame: (1616494455.806185) can0 700#A200FAFE070000
Sending MQTT Message: 700#A200FAFE070000 and Topic Can2Mqtt
```

Unfortunately, after successfully starting the application, it stops after some readings and does not continue to produce any output. I have not seen more then 15 `Received CAN Frame` messages in total throughout multiple tries. The applications seems to remain in some "idle" state without crashing or something similar. A parallel running `candump can0` continue to show messages though.

Different to your setup from the Readme, I use the PiCan2 shield, but I would not expect errors from that?!

Please advise me with any instructions which can help to debug this issue.

**Comment by maramend** [comment]
Hi everyone,

I digged into the topic and narrowed done the issue a little bit. In parallel to `can2mqtt` I connected `telnet` to the canlogserver.

It seems, the canlogserver is continously reporting frames, while `can2mqtt` stops reading.

```
Application started. Press Ctrl+C to shut down.
Hosting environment: Production
Content root path: /home/pi/can2mqtt_core/
CONNECTED TO MQTT BROKER 192.168.1.50 using ClientId Can2Mqtt
CONNECTED TO CANLOGSERVER 192.168.1.50 ON PORT 28700
Received CAN Frame: (1616610722.839816) can0 180#2200FA4F460091
Sending MQTT Message: 180#2200FA4F460091 and Topic Can2Mqtt
Received CAN Frame: (1616610722.976826) can0 100#3100FA4F410000
Received CAN Frame: (1616610722.979270) can0 180#2200FA4F41018C
Sending MQTT Message: 180#2200FA4F41018C and Topic Can2Mqtt
Received CAN Frame: (1616610723.123433) can0 100#3100FA4F430000
Received CAN Frame: (1616610723.128605) can0 180#2200FA4F430172
Sending MQTT Message: 180#2200FA4F430172 and Topic Can2Mqtt
```
```
pi@raspberrypi:~ $ telnet 127.0.0.1 28700
Trying 127.0.0.1...
Connected to 127.0.0.1.
Escape character is '^]'.
(1616610722.830291) can0 100#3100FA4F460000
(1616610722.839816) can0 180#2200FA4F460091 << First capture
(1616610722.976826) can0 100#3100FA4F410000
(1616610722.979270) can0 180#2200FA4F41018C
(1616610723.123433) can0 100#3100FA4F430000
(1616610723.128605) can0 180#2200FA4F430172 << Last capture
(1616610726.936859) can0 100#3100FA4F460000
(1616610726.939691) can0 180#2200FA4F460091
(1616610727.083568) can0 100#3100FA4F410000
(1616610727.089079) can0 180#2200FA4F41018D
(1616610727.230177) can0 100#3100FA4F430000
(1616610727.238420) can0 180#2200FA4F430172
```

**Comment by maramend** [comment]
Hi,

I figured out the issue. As you outline in https://github.com/Hunv/can2mqtt/blob/0242d609aa2f909b4c7b025cb01f9e343e20d976/can2mqtt_core/can2mqtt_core/can2mqtt.cs#L120
this expects a devicename `slcan0`, which is two character more than my devicename `can0`.

Considering this in
https://github.com/Hunv/can2mqtt/blob/0242d609aa2f909b4c7b025cb01f9e343e20d976/can2mqtt_core/can2mqtt_core/can2mqtt.cs#L112
https://github.com/Hunv/can2mqtt/blob/0242d609aa2f909b4c7b025cb01f9e343e20d976/can2mqtt_core/can2mqtt_core/can2mqtt.cs#L131
https://github.com/Hunv/can2mqtt/blob/0242d609aa2f909b4c7b025cb01f9e343e20d976/can2mqtt_core/can2mqtt_core/can2mqtt.cs#L145

and change the length to `44`, respectively `43` solves the issue. I wonder if the device name should be dynamically determined by a startup parameter, e.g. `--Daemon:CANInterface` and should also matched against the existing network interfaces in the OS to exclude errors like mine.

**Comment by maramend** [comment]
Solved by #2
